lib/_stream_readable.js

```diff
@@ -251,11 +251,7 @@ Readable.prototype._destroy = function(err, cb) {
 };
 
 Readable.prototype[EE.captureRejectionSymbol] = function(err) {
-  // TODO(mcollina): remove the destroyed if once errorEmitted lands in
-  // Readable.
-  if (!this.destroyed) {
-    this.destroy(err);
-  }
```

Comment:
@mcollina: FYI. With `autoDestroy`, `push(null)` will cause the stream to be destroyed and thus the error doesn't propagate (failing a test). Since your comment has landed I believe removing the condition is safe.

Reply:
agreed